extensions/amp-bind/0.1/bind-expression.js

```diff
@@ -83,8 +93,12 @@ export class BindExpression {
     /** @const {string} */
     this.expressionString = expressionString;
 
-    /** @const {!./bind-expr-defines.AstNode} */
+    /** @const @private {!./bind-expr-defines.AstNode} */
     this.ast_ = parser.parse(this.expressionString);
+
+    /** @const @private {!Object} */
+    this.builtInScope_ = Object.create(null);
+    this.builtInScope_['Math'] = Math;
```

Comment:
I am still concerned about this. Requiring `state` definition be aware of a list of reserved keywords is not a good idea. In real-world scenarios, state is normally not handwritten, it is auto-generated from some data and a lot of times that data itself is based on user-input. Requiring the developer to sanitize the state to account for our reserved keywords is not great.

Imagine a scenario where server-side template uses _username_ as amp-bind variables:
```
// this is a server-size template,`{{Username}}` will be replaced by a JSEncoded string of the username at render time
<button on="tap:AMP.setState({{Username}}, true)>Activate your account</button>

<div [show]="{{Username}}">
 You account is active.
 Your lucky number is <span [text]="Math.random()"></span>
</div>
```

Now with this code everything is fine until someone decides to use `Math` as their username causing a user-specific, hard-to-debug issue.

Maybe the right solution is clearly separating builtin functions from state and state property access syntax. e.g. `-20 | Math.floor` instead of `Math.floor(-20)` this way, `Math` is not longer a `scope` same way `state` is a scope.